### PR TITLE
add pr-thanks-by-sendgrid.rs

### PR DESCRIPTION
**Pull request by ghost**
update README.md accordingly

### CODE REVIEW

**Review by ho-229 — CHANGES_REQUESTED** [review]
Can you implement it with [connector-dsi](https://github.com/second-state/connector-dsi).
You can refer to <https://github.com/second-state/flow-functions/blob/main/github/sendgrid/star-thanks-by-sendgrid/rust/src/star-thanks-by-sendgrid.rs>

**Comment by ghost** [comment]
updated script to new syntax.
brutally committed to the webpage version due to my lack of experience with Git, pls watch out possible problems for me, many thanks!

**Comment by ghost** [comment]
re-opened pr

**Comment by ghost** [comment]
need to change a line

**Comment by ghost** [comment]
changed "pull_request" check logic

File: github/sendgrid/pr-thanks-by-sendgrid/rust/src/pr-thanks-by-sendgrid.rs [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    _run(s)
+}
+
+fn _run(s: String) -> Result<String, String> {
+    let payload = inbound(s)?;
+    
+    match payload.pull_request {
+        None => { return Err("no pull request found".to_string()); },
+        Some(_) => {},
+    }
```
Comment (ho-229) [inline github/sendgrid/pr-thanks-by-sendgrid/rust/src/pr-thanks-by-sendgrid.rs]: In fact `pull_request` field is unnecessary in your program.
btw, if you want to check whether `pull_request` is exists, you should use:

```rust
payload.get_pull_request()?;
```

`get_xxx()` will return a `Result` which including the error message, and than `?` operator will return when it was 'Err()'

File: github/sendgrid/pr-thanks-by-sendgrid/rust/src/pr-thanks-by-sendgrid.rs [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+    let mut subject: &str = "";
+    let mut content = String::new();
+
+    if payload.action.unwrap() == "opened" {
```
Comment (ho-229) [inline github/sendgrid/pr-thanks-by-sendgrid/rust/src/pr-thanks-by-sendgrid.rs]: A better implementation is return early when the condition is not met. It should like:

```rust
if payload.get_action()? != "opened" {
    return Ok(String::new());
}
```

**Review by ho-229 — APPROVED** [review]

**Review by ho-229 — CHANGES_REQUESTED** [review]
sorry it's change request

File: github/sendgrid/pr-thanks-by-sendgrid/rust/src/pr-thanks-by-sendgrid.rs (hunk 2 above)
Comment (ghost) [reply github/sendgrid/pr-thanks-by-sendgrid/rust/src/pr-thanks-by-sendgrid.rs]: thanks for your suggestions

File: github/sendgrid/pr-thanks-by-sendgrid/rust/src/pr-thanks-by-sendgrid.rs (hunk 1 above)
Comment (ghost) [reply github/sendgrid/pr-thanks-by-sendgrid/rust/src/pr-thanks-by-sendgrid.rs]: nice

File: github/sendgrid/pr-thanks-by-sendgrid/rust/src/pr-thanks-by-sendgrid.rs [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+            r#"
+Hi {}, <br/>
+
+Welcome to the {} community, thank you for your contribution!
+                        "#,
+            payload.sender.login, payload.repository.unwrap().full_name
+        );
+    
+
+    Ok(json!([{
```
Comment (ho-229) [inline github/sendgrid/pr-thanks-by-sendgrid/rust/src/pr-thanks-by-sendgrid.rs:32]: Actually you can use `flows_connector_dsi::sendgrid::outbound` here if you like😂

**Review by ho-229 — APPROVED** [review]

**Pull request merged**